src/implementations/vanilla/components/tabs/tabs.js

```diff
@@ -334,6 +374,100 @@ export class Tabs {
     this.moreButton.setAttribute('aria-expanded', false);
     this.dropdown.classList.remove('ecl-tabs__dropdown--show');
   }
+
+  /**
+   * Tabs onKeydown handler.
+   * @param {Event} e
+   */
+  onKeydown(e) {
+    const tgt = e.currentTarget;
+    let flag = false;
+
+    switch (e.key) {
+      case 'ArrowLeft':
+        this.moveFocusToPreviousTab(tgt);
+        flag = true;
+        break;
+
+      case 'ArrowRight':
+        this.moveFocusToNextTab(tgt);
+        flag = true;
+        break;
+
+      case 'Home':
+        this.moveFocusToTab(this.firstTab);
+        flag = true;
+        break;
+
+      case 'End':
+        this.moveFocusToTab(this.lastTab);
+        flag = true;
+        break;
+
+      default:
+        break;
+    }
+
+    if (flag) {
+      e.stopPropagation();
+      e.preventDefault();
+    }
+  }
+
+  /**
+   * @param {HTMLElement} currentTab tab element
+   */
+  moveFocusToTab(currentTab) {
+    if (currentTab.closest('.ecl-tabs__dropdown')) {
+      this.dropdown.classList.add('ecl-tabs__dropdown--show');
+    } else {
+      this.dropdown.classList.remove('ecl-tabs__dropdown--show');
+    }
+    currentTab.focus();
+    return this;
+  }
+
+  /**
+   * @param {HTMLElement} currentTab tab element
+   */
+  moveFocusToPreviousTab(currentTab) {
```

Comment:
Do we really need separated callbacks foran operation like moving focus between the tags..? It's only only because of the almost duplicated code, but it's about the logic, i feel that we could handle this in a single function, but it's just a suggestion and a preference..

Reply:
Merged the 2 functions